### PR TITLE
Update App Header & Banner

### PR DESCRIPTION
**Pull request by Nitemaeric**
## Context

- We want to be able to configure the hosting environment separately from the Rails environment.
  - This is to create a distinction between configuring Rails and configuring a hosted environment.
- We want to add colors to the app's header bottom border and phase banner tag background based on the current hosting environment.

As I was going through the changes, I also found some bugs and opportunities to improve what was closely related.

- Clicking on the Logo breaks the Claims app when logged in as a support user.
- The Phase Banner text didn't match the prototype.
- The feedback link in the Phase Banner didn't go anywhere.
- The existing overridden styles for the header did not support the mobile menu.

## Changes proposed in this pull request

- Add a `HOSTING_ENV` env var to control which environment we're targeting.
- Update header bottom border to be full width.
- Update `HostingEnvironment.env` to provide a matching experience to `Rails.env`.
  - [ActiveSupport::EnvironmentInquirer](https://www.rubydoc.info/github/rails/rails/ActiveSupport/EnvironmentInquirer)
- Pull out the phase banner localisation from `HostingEnvironment`.
- Update phase banner text to match prototype.
- Clicking on the header logo should take you to the `sign_in_path` or the user's `after_sign_in_path`.
- Add a _placeholder_ feedback page.
- Add [Climate Control](https://github.com/thoughtbot/climate_control) to safely set and reset ENV vars in tests.

## Guidance to review

- Try running the app with `HOSTING_ENV=production` and etc.
- By default, you should see the header and tags in grey.
- The header navigation menu is now properly full width on mobile.
- Clicking on the header's logo should redirect to path of least surprise now.

## Screenshots

| `HOSTING_ENV` | Screenshot |
| ----------------- | ------------ |
| production        | CleanShot 2024-01-30 at 16 15 34 |
| staging | CleanShot 2024-01-30 at 16 16 46 |
| sandbox | CleanShot 2024-01-30 at 16 17 13 |
| qa | CleanShot 2024-01-30 at 16 17 36 |
| review | CleanShot 2024-01-30 at 16 18 00 |
| development | CleanShot 2024-01-30 at 16 18 28 |

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -99,6 +99,7 @@ group :test do
   # The accessible selectors gem is maintained by Citizens Advice, not yet a registered gem.
   gem "capybara_accessible_selectors", git: "https://github.com/citizensadvice/capybara_accessible_selectors", branch: "main"
   gem "capybara-screenshot"
+  gem "climate_control"
```
[the remark this answers is not in the document]
Comment (Nitemaeric) [reply Gemfile:102]: > - Add [Climate Control](https://github.com/thoughtbot/climate_control) to safely set and reset ENV vars in tests.

Added to PR description. 😄

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -25,24 +25,20 @@
...
-      <div class="govuk-width-container">
-        <%= govuk_phase_banner(tag: { text: HostingEnvironment.name(current_service) }, classes: [class_names("govuk-phase-banner--no-border-bottom": support_controller?)]) do %>
-          <%= govuk_link_to HostingEnvironment.banner_description(current_service), request.env["PATH_INFO"], class: "govuk-link--no-visited-state" %>
-        <% end %>
-      </div>
-    </div>
-
     <div class="govuk-width-container">
+      <%= govuk_phase_banner(tag: { text: HostingEnvironment.phase(current_service).upcase }, classes: [class_names("app-phase-banner", "app-phase-banner__env--#{HostingEnvironment.env}", "app-phase-banner--no-border-bottom": support_controller?)]) do %>
+        <%== t(".#{current_service}.phase_banner.description", feedback_link: govuk_link_to(t(".#{current_service}.phase_banner.feedback"), feedback_path, class: "govuk-link--no-visited-state")) %>
```
[the remark this answers is not in the document]
Comment (Nitemaeric) [reply app/views/layouts/application.html.erb:39]: Do you have a screenshot?

**Comment by Nitemaeric** [comment]
[[Bug] Clicking on the banner logo should redirect Support Users to /support](https://trello.com/c/wLodLNxG/155-bug-clicking-on-the-banner-logo-should-redirect-support-users-to-support)

**Comment by Nitemaeric** [comment]
[Add environment colors to App Header & Update App Header width](https://trello.com/c/GQYWHC9H/154-add-environment-colors-to-app-header-update-app-header-width)

**Comment by simonwhatley** [comment]
The production version should use the default GOV.UK phase banner tag style:

Screenshot 2024-01-30 at 15 58 41@2x

```
<div class="govuk-phase-banner">
  <p class="govuk-phase-banner__content">
    <strong class="govuk-tag govuk-phase-banner__content__tag">
      Beta
    </strong>
    <span class="govuk-phase-banner__text">
      This is a new service – your <a class="govuk-link" href="#">feedback</a> will help us to improve it.
    </span>
  </p>
</div>
```

File: app/assets/stylesheets/components/_header.scss [hunk 3]
```diff
@@ -1,8 +1,43 @@
...
+      border-bottom-color: govuk-colour("blue");
+    }
+
+    &.app-header--qa {
+      border-bottom-color: govuk-colour("orange");
+    }
+
+    &.app-header--staging {
+      border-bottom-color: govuk-colour("red");
+    }
```
Comment (ollietreend) [inline app/assets/stylesheets/components/_header.scss:36]: Would it make sense to name these classes according to their colours, and then use the existing `env => colour` mapping defined in `PhaseBannerHelper`?

```suggestion
    &.app-header--blue {
      border-bottom-color: govuk-colour("blue");
    }

    &.app-header--orange {
      border-bottom-color: govuk-colour("orange");
    }

    &.app-header--red {
      border-bottom-color: govuk-colour("red");
    }
```

That way we'll have one single source of truth for environment colours:

https://github.com/DFE-Digital/itt-mentor-services/blob/279f3db10905c878e7b6810acfc835c4b0064f29/app/helpers/phase_banner_helper.rb#L2-L8

And we can use the new `phase_banner_color` helper method in the view:

```ruby
<%= govuk_header(homepage_url: "/", classes: "app-header app-header--#{phase_banner_color} app-header--full-border") do |header| %>
```

File: lib/hosting_environment.rb [hunk 4]
```diff
@@ -4,15 +4,17 @@ module HostingEnvironment
...
+  def self.phase(current_service)
+    if env.production?
       PRODUCTION_BANNER_NAME.fetch(current_service)
+    elsif env.qa?
+      "QA"
     else
-      Rails.env
+      env
     end
   end
```
Comment (ollietreend) [inline lib/hosting_environment.rb:19]: This `HostingEnvironment.phase` method still looks & smells like a View concern to me.

Would it sit more comfortably in the `PhaseBannerHelper`?

**Review by elceebee — APPROVED** [review]
LGTM! 🚀

File: app/views/claims/pages/feedback.html.erb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<% content_for(:page_title) { t(".page_title") } %>
+
+<div class="govuk-width-container">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <h1 class="govuk-heading-l"><%= t(".page_title") %></h1>
+    </div>
+  </div>
+</div>
```
Comment (ollietreend) [inline app/views/claims/pages/feedback.html.erb:9]: Do we know for sure that a standalone feedback page is required?

I'm used to seeing the "feedback" link open an external survey/feedback form. But I expect @simonwhatley will have an opinion on this.

**Review by ollietreend — COMMENTED** [review]
This is looking good to me.

No blockers, but I do have a few questions & suggestions.

File: app/views/claims/pages/feedback.html.erb (hunk 5 above)
Comment (Nitemaeric) [reply app/views/claims/pages/feedback.html.erb:9]: The last we spoke about this topic, I think we were going to take the simplest route, similar to [apply's feedback page](https://www.apply-for-teacher-training.service.gov.uk/candidate/make-a-complaint).

Ref: https://ukgovernmentdfe.slack.com/archives/C06962474AC/p1706277019469919?thread_ts=1706274737.868929&cid=C06962474AC

Comment (Nitemaeric) [reply app/views/claims/pages/feedback.html.erb:9]: It's also very easy to drop this if we don't need it.

File: lib/hosting_environment.rb (hunk 4 above)
Comment (Nitemaeric) [reply lib/hosting_environment.rb:19]: The values themselves are data, the "QA" string manipulation is a valid View concern though. I've moved that into a view helper.

**Review by ollietreend — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**